src/react-components/room/RoomEntryModal.js

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import { Modal } from "../modal/Modal";
+import { Button } from "../input/Button";
+import { ReactComponent as EnterIcon } from "../icons/Enter.svg";
+import { ReactComponent as VRIcon } from "../icons/VR.svg";
+import { ReactComponent as ShowIcon } from "../icons/Show.svg";
+import { ReactComponent as SettingsIcon } from "../icons/Settings.svg";
+import styles from "./RoomEntryModal.scss";
+import styleUtils from "../styles/style-utils.scss";
+import { useCssBreakpoints } from "react-use-css-breakpoints";
+
+export function RoomEntryModal({
+  appName,
+  logoSrc,
+  className,
+  roomName,
+  onJoinRoom,
+  onEnterOnDevice,
+  onSpectate,
+  onOptions,
+  ...rest
+}) {
+  const breakpoint = useCssBreakpoints();
+  return (
+    <Modal
+      className={classNames(styles.roomEntryModal, className)}
+      contentClassName={styles.content}
+      disableFullscreen
+      {...rest}
+    >
+      {breakpoint !== "sm" && (
+        <div className={styles.logoContainer}>
+          <img src={logoSrc} alt={appName} />
+        </div>
+      )}
+      <div className={styles.roomName}>
+        <h5>Room Name</h5>
+        <p>{roomName}</p>
+      </div>
+      <div className={styles.buttons}>
+        <Button preset="blue" onClick={onJoinRoom}>
+          <EnterIcon /> Join Room
+        </Button>
+        <Button preset="purple" onClick={onEnterOnDevice}>
+          <VRIcon /> Enter On Device
+        </Button>
+        <Button preset="orange" onClick={onSpectate}>
+          <ShowIcon /> Spectate
+        </Button>
```

Comment:
I think it would be good to emphasis "Join Room" more than "Enter On Device" and "Spectate". Seems like making this first choice easy will help newcomers (who may not have any conception of what a room is, what enter on device means, or what spectate does), and even beyond that, I suspect "Join Room" is the most common action by a _significant_ margin.
![image](https://user-images.githubusercontent.com/4072106/93265804-fd2faa00-f75d-11ea-895c-9cbc537c133d.png)

Reply:
I'm also curious to see how this would look if the icons were vertically aligned with on another